### PR TITLE
[#1491] feat(trino-connector): Support query optimize in the Trino-connector

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

- Support for query optimization interfaces in the Trino-connector includes projection, aggregation, predicate, TopN, and join pushdown.
- Add a version to the creation of the internal connector to avoid `Class loader` name conflicts when recreating the catalog

### Why are the changes needed?

Feature: #1491 

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Add tester in testset

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java [hunk 1]
```diff
@@ -325,4 +363,175 @@ public void setColumnComment(
...
+  public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
+      ConnectorSession session,
+      JoinType joinType,
+      ConnectorTableHandle left,
+      ConnectorTableHandle right,
+      ConnectorExpression joinCondition,
+      Map<String, ColumnHandle> leftAssignments,
+      Map<String, ColumnHandle> rightAssignments,
+      JoinStatistics statistics) {
+    if (!(left instanceof GravitinoTableHandle) && !(right instanceof GravitinoTableHandle)) {
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java:377]: The condition should be `!(left instanceof GravitinoTableHandle) || !(right instanceof GravitinoTableHandle)`?

**Comment by diqiu50** [comment]
@jerryshao Please take a review

**Comment by jerryshao** [comment]
@yuqi1129 can you please help to review first?

**Comment by yuqi1129** [comment]
> @yuqi1129 can you please help to review first?

In fact, I have reviewed it already, anyway, I will review it again.

File: integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-postgresql/00004_query_pushdown.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+"Trino version: %
+%
+    └─ TableScan[table = test.gt_postgresql:gt_db1.customer gt_db1.customer limit=10 columns=[custkey:bigint:int8]]
+           Layout: [custkey:bigint]
+%
+"
+
+"Trino version: %
+%
+    └─ TableScan[table = test.gt_postgresql:gt_db1.customer gt_db1.customer constraints=[ParameterizedExpression[expression=(""phone"") LIKE (?), parameters=[QueryParameter{jdbcType=Optional.empty, type=varchar(6), value=Optional[Slice{base=[B@%, baseOffset=0, length=6}]}]]] limit=10]
```
Comment (yuqi1129) [inline integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-postgresql/00004_query_pushdown.txt:22]: How can we know the impact of a condition that has been pushed down?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/jdbc/postgresql/PostgreSQLConnectorAdapter.java [hunk 3]
```diff
@@ -25,7 +26,8 @@ public PostgreSQLConnectorAdapter() {
   public Map<String, Object> buildInternalConnectorConfig(GravitinoCatalog catalog)
       throws Exception {
     Map<String, Object> config = new HashMap<>();
-    config.put("catalogHandle", catalog.getName() + ":normal:default");
+    config.put(
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/jdbc/postgresql/PostgreSQLConnectorAdapter.java:29]: Can you add some content to the PR about this changes?

File: integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-postgresql/00004_query_pushdown.txt (hunk 2 above)
Comment (diqiu50) [reply integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-postgresql/00004_query_pushdown.txt:22]: If predicate pushdown for a specific clause is succesful, the EXPLAIN plan for the query does not include a ScanFilterProject operation for that clause.

https://trino.io/docs/current/optimizer/pushdown.html#predicate-pushdown

Comment (yuqi1129) [reply integration-test/src/test/resources/trino-ci-testset/testsets/jdbc-postgresql/00004_query_pushdown.txt:22]: I see.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by yuqi1129** [comment]
@jerryshao 
Do you have any further comments?

**Comment by jerryshao** [comment]
You can go ahead if you think it is OK.

**Pull request merged**